components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java

```diff
@@ -964,26 +997,38 @@ private static boolean isResourceEpUnavailable(Operation operation) {
         return false;
     }
 
-    public static List<String> getAuthProviders(String schemas) {
+    public static List<String> getAuthProviders(String schemas, List<String> apiSecurityByExtension) {
         List<String> authProviders = new ArrayList<>();
-        boolean basic = false;
-        boolean oauth2 = false;
-        if (schemas != null) {
-            String[] schemasArray = schemas.trim().split("\\s*,\\s*");
-            for (String s : schemasArray) {
-                if (s.equalsIgnoreCase(APISecurity.basic.name())) {
-                    authProviders.add(APISecurity.basic.name());
-                } else if (s.equalsIgnoreCase(APISecurity.apikey.name())) {
-                    authProviders.add(APISecurity.apikey.name());
-                } else if (s.equalsIgnoreCase(APISecurity.oauth2.name())) {
-                    authProviders.add(APISecurity.oauth2.name());
-                    authProviders.add(APISecurity.jwt.name());
+        // Support api manager application level security
+        if (apiSecurityByExtension != null && !apiSecurityByExtension.isEmpty()) {
+            for (String securityType : apiSecurityByExtension) {
+                if (OpenAPIConstants.APPLICATION_LEVEL_SECURITY.containsKey(securityType)) {
+                    setDefaultSecurityScheme(
+                            OpenAPIConstants.APPLICATION_LEVEL_SECURITY.get(securityType), authProviders);
                 }
             }
+        } else if (schemas != null) {
+            String[] schemasArray = schemas.trim().split("\\s*,\\s*");
+            for (String securityType : schemasArray) {
+                setDefaultSecurityScheme(securityType, authProviders);
+            }
         }
+
         if (authProviders.isEmpty()) {
-            authProviders.add(APISecurity.oauth2.name());
-            authProviders.add(APISecurity.jwt.name());
+            authProviders.add(OpenAPIConstants.APISecurity.oauth2.name());
+            authProviders.add(OpenAPIConstants.APISecurity.jwt.name());
+        }
+        return authProviders;
+    }
+
+    private static List<String> setDefaultSecurityScheme(String securityType, List<String> authProviders) {
```

Comment:
`setAuthProviders()`???

Reply:
Fixed in https://github.com/wso2/product-microgateway/pull/970/commits/c023abac6801c1fdbfbeb4795dac35a90ea8b045